packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx

```diff
@@ -78,6 +78,14 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
           .getLocaleText(`filterOperator${capitalize(item.operatorValue!)}` as GridTranslationKeys)!
           .toString();
 
+      const getFilterItemValue = (item: GridFilterItem): string => {
+        const { getValueAsString } = lookup[item.columnField!].filterOperators!.find(
+          (operator) => operator.value === item.operatorValue,
+        )!;
+
+        return getValueAsString ? getValueAsString(item.value) : item.value;
+      };
```

Comment:
Filter item got [updated](https://github.com/mui/mui-x/pull/6985), can you rebase with latest `next` ?
```suggestion
      const getFilterItemValue = (item: GridFilterItem): string => {
        const { getValueAsString } = lookup[item.field!].filterOperators!.find(
          (operator) => operator.value === item.operator,
        )!;

        return getValueAsString ? getValueAsString(item.value) : item.value;
      };
```

Reply:
All done.